### PR TITLE
[Hotfix][Format][Canal-Json]  Fix json deserialize NPE

### PR DESCRIPTION
**Pull request by zhilinli123**
…ll pointer prevention

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
Use `mvn spotless:apply` format code and recommit. Thanks

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java [hunk 1]
```diff
@@ -109,24 +109,23 @@ public SeaTunnelDataType<SeaTunnelRow> getProducedType() {
...
-        if (table != null) {
-            if (!tablePattern.matcher(jsonNode.get(FIELD_TABLE).asText()).matches()) {
-                return;
-            }
+        if (table != null && !tablePattern.matcher(jsonNode.get(FIELD_TABLE).asText()).matches()) {
+            return;
         }
         JsonNode dataNode = jsonNode.get(FIELD_DATA);
         String type = jsonNode.get(FIELD_TYPE).asText();
+        assert dataNode != null;
```
Comment (liugddx) [inline seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java]: Can a business exception be thrown?

Comment (zhilinli123) [reply seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java:128]: > 是否可以抛出业务异常？



> 是否可以抛出业务异常？
Like this?
if (dataNode == null) {
            new NullPointerException("deserialize cancal-json ‘data’ is null" + jsonNode)
        }

Comment (zhilinli123) [reply seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java:128]: > Can a business exception be thrown?

Like this?
`if (dataNode == null) {
new NullPointerException("deserialize cancal-json ‘data’ is null" + jsonNode)
}`

Comment (zhilinli123) [reply seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java]: Like this?
if (dataNode == null) { 
throw new NullPointerException("deserialize cancal-json ‘data’ is null" + jsonNode);
}

**Comment by zhilinli123** [comment]
> Use `mvn spotless:apple` format code and recommit. Thanks

`mvn spotless:apply`？

**Comment by TyrantLucifer** [comment]
> > Use `mvn spotless:apple` format code and recommit. Thanks
> 
> `mvn spotless:apply`？

Sorry my fault. You are right.

File: seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java (hunk 1 above)
Comment (TyrantLucifer) [reply seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java]: [throw new SeaTunnelJsonFormatException()](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/exception/SeaTunnelJsonFormatException.java)

Comment (zhilinli123) [reply seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java]: please review

Comment (zhilinli123) [reply seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java]: > please review
Case:
image

**Comment by TaoZex** [comment]
An exception occurs when run this line of code. Please fix it.

**Comment by TyrantLucifer** [comment]
CI/CD has restarted. cc @MonsterChenzhuo PTAL

**Comment by MonsterChenzhuo** [comment]
You need to run the test case locally. There is a test case with a null value that can't pass.

**Comment by zhilinli123** [comment]
> You need to run the test case locally. There is a test case with a null value that can't pass.

There's a null test in the E2E module and I've removed it for now and when that happens we'll throw an exception to alert users @TyrantLucifer @MonsterChenzhuo @hailin0

**Comment by zhilinli123** [comment]
> > You need to run the test case locally. There is a test case with a null value that can't pass.
> 
> There's a null test in the E2E module and I've removed it for now and when that happens we'll throw an exception to alert users CC @TyrantLucifer @MonsterChenzhuo @hailin0

**Comment by zhilinli123** [comment]
> You need to run the test case locally. There is a test case with a null value that can't pass.
Can you help me with this CI problem 
E2e now collects data of type Query  @MonsterChenzhuo

**Comment by EricJoy2048** [comment]
Please fix the ci error.

**Comment by zhilinli123** [comment]
> Please fix the ci error.
done ～

**Comment by zhilinli123** [comment]
flink skipped the ddl event and I think we should throw an exception to prompt the user

```
 else if (OP_CREATE.equals(type)) {
                // "data" field is null and "type" is "CREATE" which means
                // this is a DDL change event, and we should skip it.
                return;
            }
```

**Comment by zhilinli123** [comment]
The bosses help to see if this meets expectations

**Review by EricJoy2048 — APPROVED** [review]

**Comment by TyrantLucifer** [comment]
@MonsterChenzhuo PTAL

**Comment by MonsterChenzhuo** [comment]
LGTM

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**